### ISSUE TITLE
Check colours for accessibility / WCAG compliance

### ISSUE THREAD

**Issue by rixx** [django/djangoproject.com#974]
There are a couple of areas where the current colour contrasts between fonts and backgrounds fail the WCAG 2.0 colour contrast accessibility guidelines on the djangoproject.com pages.
It's mostly to do with the bright green compared with white/white-ish colours:

### Green on gray, [contrast 2.79](https://contrast-ratio.com/#rgb%2832%2C%20170%2C%20118%29-on-%23f8f8f8)

Fails all checks, mostly in use on sidebars.
![DeepinScreenshot_select-area_20200129111742](https://user-images.githubusercontent.com/2657547/73348051-fadd1600-4288-11ea-84ec-879e66c8d283.png)

### Bright green on less bright green, [contrast 2.02](https://contrast-ratio.com/#%23C9F0DD-on-%2344b78b)

Fails all checks, it's in use in title areas.
![DeepinScreenshot_select-area_20200129111856](https://user-images.githubusercontent.com/2657547/73348168-32e45900-4289-11ea-9683-c42a45fed94b.png)

### Bright green on dark green, [contrast 4.05](https://contrast-ratio.com/#%2344B78B-on-%230c4b33)

This would be okay with 14pt+ instead of the current ~10pt, but 14pt is unreasonably huge.

![DeepinScreenshot_select-area_20200129112047](https://user-images.githubusercontent.com/2657547/73348417-a9815680-4289-11ea-9ed9-7f1574c41a31.png)

### Green on green, [contrast 2.43](https://contrast-ratio.com/#rgb%2843%2C%20140%2C%20103%29-on-%230c4b33)

Fails all checks, is hard to read even for me.

![DeepinScreenshot_select-area_20200129112333](https://user-images.githubusercontent.com/2657547/73348570-ebaa9800-4289-11ea-92cd-022cdbf994bb.png)

### White on green, [contrast 2.5](https://contrast-ratio.com/#white-on-%2344B78B)

Sadly, this isn't even WCAG compliant in large/bold fonts, and we use it on all of our buttons, etc.
![DeepinScreenshot_select-area_20200129112526](https://user-images.githubusercontent.com/2657547/73348732-34625100-428a-11ea-8d23-12a47f1a10d0.png)

## Next steps

For me (good vision, good screen) some of the completely failing combinations are perfectly easy to read, like the white font on bright green buttons – but that's the point of accessibility guidelines, they're not for people with good vision and good screens.

Since this impacts a lot of the djangoproject.com page, especially with the bright green being so prevalent, I'm not sure how to proceed here, but I wanted to document the issue to start a conversation about it.

**Comment by ryanwilsonperkin** [comment]
Looks like switching to using `color: white` for text on the green backdrops would be a relatively quick win here. It has an acceptable contrast ratio on both `#44B78B` which is used for the `.copy-banner` and on `#0C4B33` which is used for the navigation banner on the top.

**Comment by rixx** [comment]
> It has an acceptable contrast ratio on both `#44B78B`

How did you check the contrast? The last example is exactly white on `#44B78B`, and the [contrast is 2.5](https://contrast-ratio.com/#white-on-%2344B78B), which doesn't even pass in large/bold fonts.

**Comment by ryanwilsonperkin** [comment]
Interesting, I'd tried on http://colorsafe.co/ and it seemed to think the contrast ratio was much higher. My mistake

**Comment by thibaudcolas** [comment]
There are a lot of shades of green used on the site currently, but it seems like it’s only the intermediary "bright greens" that are problematic:

<img width="830" src="https://user-images.githubusercontent.com/877585/97096172-2ba77b80-1660-11eb-9fc3-a7133e58e3a5.png" alt="Shades of green for background color, text color, fill, borders on djangoproject.com">

I’d recommend replacing their usage with a new darker shade. Quick tests with Leonardo: https://leonardocolor.io/?colorKeys=%230c4b33%2C%2344b78b&base=c9f0dd&ratios=4.6%2C7&mode=CAM02. That would be `#247454` to have something that works with `#C9F0DD` as the text color.

For "bright green" for text,

- On dark backgrounds it could be replaced with white or `#C9F0DD`.
- On light backgrounds it could be replaced by this new shade, `#247454`.

---

Here is a mockup of the result:

![www djangoproject com_start_overview_](https://user-images.githubusercontent.com/877585/97096284-d9fff080-1661-11eb-9df7-b3c29bf8ddc7.png)

New color palette:

<img width="549" alt="djangoproject-colors-after" src="https://user-images.githubusercontent.com/877585/97096295-03208100-1662-11eb-9dc1-68956aaddbd7.png">

**Comment by CuriousLearner** [comment]
Hello @rixx 

Can you please let us know why this issue was closed? Apparently there is a PR #998 opened for it.

**Comment by rixx** [comment]
I don't recall why I closed this – I may have assumed that this was either merged or wasn't going to get merged two years after not seeing any movement on it. Happy to reopen!

**Comment by knyghty** [comment]
@thibaudcolas on the docs site there are also red (for some reason?) links on the docs pages: https://docs.djangoproject.com/en/4.1/ - not sure if you want to add those to your palette.

**Comment by kalafut** [comment]
I'd like to add another example that I was going to report before ending up on this issue. Here's the search box, which is barely readable in light mode:

<img width="398" alt="image" src="https://user-images.githubusercontent.com/211880/196529795-7f6b9434-1bb6-4b3f-a189-22103b7c9ea0.png">

**Comment by akshatvermavi** [comment]
Looks like switching to using color: white for text on the green backdrops would be a relatively quick win here. It has an acceptable contrast ratio on both #FFFFFF which is used for text and on #0C4B33 which is used for background color.

**Comment by thibaudcolas** [comment]
[Current color contrast grid for future reference](https://contrast-grid.eightshapes.com/?version=1.1.0&background-colors=%23ffffff%0D%0A%23f1fff7%0D%0A%23f8f8f8%0D%0A%2344b78b%0D%0A%2320aa76%0D%0A%23187f58%0D%0A%230c4b33&foreground-colors=%23ffffff%0D%0A%23f1fff7%0D%0A%23c9f0dd%0D%0A%2344b78b%0D%0A%23859d94%0D%0A%2320aa76%0D%0A%23798780%0D%0A%232b8c67%0D%0A%230c4b33%0D%0A%230c3c26%0D%0A%23000000&es-color-form__tile-size=compact&es-color-form__show-contrast=aaa&es-color-form__show-contrast=aa)

<img width="1082" alt="current color contrast grid for website" src="https://github.com/django/djangoproject.com/assets/877585/4affbebd-cb49-4670-b97f-5ab42d96456b">

**Comment by kalafut** [comment]
I've recently noticed another area that could use improvement. The syntax highlighting of code blocks uses very light grey for braces/brackets/parens/commas/etc.:

<img width="455" alt="image" src="https://github.com/user-attachments/assets/5ca5b489-38fb-4997-b0a9-1d8707e78fdb">

These are about a 1.45 contrast.
